lib/reporting/identity_verification_report.rb

```diff
@@ -0,0 +1,239 @@
+        csv << ['Incomplete Users', incomplete_users]
+        csv << ['Address Confirmation Letters Requested', idv_gpo_address_letter_requested]
+        csv << ['Started In-Person Verification', usps_ipp_enrollment_created]
+        csv << ['Alternative Process Users', alternative_process_users]
+        csv << ['Success through Online Verification', idv_final_resolution]
+        csv << ['Success through Address Confirmation Letters', gpo_verification_submitted]
+        csv << ['Success through In-Person Verification', usps_enrollment_status_updated]
+        csv << ['Successfully Verified Users', successfully_verified_users]
+      end
+    end
+
+    def incomplete_users
+      idv_doc_auth_image_vendor_submitted - successfully_verified_users - alternative_process_users
+    end
+
+    def idv_gpo_address_letter_requested
+      data[Events::IDV_GPO_ADDRESS_LETTER_REQUESTED].to_i
+    end
+
+    def usps_ipp_enrollment_created
+      data[Events::USPS_IPP_ENROLLMENT_CREATED].to_i
+    end
+
+    def alternative_process_users
+      [
+        idv_gpo_address_letter_requested,
+        usps_ipp_enrollment_created,
+        -gpo_verification_submitted,
+        -usps_enrollment_status_updated,
+      ].sum
+    end
+
+    def idv_final_resolution
+      data[Events::IDV_FINAL_RESOLUTION].to_i
+    end
+
+    def gpo_verification_submitted
+      data[Events::GPO_VERIFICATION_SUBMITTED].to_i
+    end
+
+    def usps_enrollment_status_updated
+      data[Events::USPS_ENROLLMENT_STATUS_UPDATED].to_i
+    end
+
+    def successfully_verified_users
+      idv_final_resolution + gpo_verification_submitted + usps_enrollment_status_updated
+    end
+
+    def idv_doc_auth_image_vendor_submitted
+      data[Events::IDV_DOC_AUTH_IMAGE_UPLOAD].to_i
+    end
+
+    # Turns query results into a hash keyed by event name, values are a count of unique users
+    # for that event
+    # @return [Hash<String,Integer>]
+    def data
+      @data ||= begin
+        event_users = Hash.new do |h, uuid|
+          h[uuid] = Set.new
+        end
+
+        # IDEA: maybe there's a block form if this we can do that yields results as it loads them
+        # to go slightly faster
+        fetch_results.each do |row|
+          event_users[row['name']] << row['user_id']
+        end
+
+        event_users.transform_values(&:count)
+      end
+    end
+
+    def fetch_results
+      cloudwatch_client.fetch(query:, from:, to:)
+    end
+
+    # @return [Time]
+    def from
+      date.in_time_zone('UTC').beginning_of_day
+    end
+
+    # @return [Time]
+    def to
+      date.in_time_zone('UTC').end_of_day
+    end
+
+    def query
+      params = {
+        issuer: quote(issuer),
+        event_names: quote(Events.all_events),
+        usps_enrollment_status_updated: quote(Events::USPS_ENROLLMENT_STATUS_UPDATED),
+        gpo_verification_submitted: quote(Events::GPO_VERIFICATION_SUBMITTED),
+      }
+
+      format(<<~QUERY, params)
```

Comment:
Um I guess it's mostly a matter of style. I like using the hash substitution syntax from `format` because it means the things we substitute in need to come from the hash, and then it's easy to make sure all the items in the hash have been quoted, it would be a little longer and a little verbose to use `#{quote(Events:: USPS_ENROLLMENT_STATUS_UPDATED)}` in the heredoc syntax